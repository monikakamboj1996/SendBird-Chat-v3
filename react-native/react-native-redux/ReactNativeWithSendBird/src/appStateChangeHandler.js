export default (function() {
  var instance;

  function AppStateChangeHandler() {
    this.cbs = {};
    this.addCallback = (key, cb) => {
      this.cbs[key] = cb;
      return () => {
        delete this.cbs[key];
      };
    };

    this.notify = () => {
      for (let key in this.cbs) {
        this.cbs[key]();
      }
    };
  }

  return {
    getInstance: function() {
      if (!instance) {
        instance = new AppStateChangeHandler();
      }
      return instance;
    }
  };
})();
