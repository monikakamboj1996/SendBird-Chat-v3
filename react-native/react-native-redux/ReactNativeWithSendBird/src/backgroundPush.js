import firebase from 'react-native-firebase';

export default async message => {
  try {
    const text = message.data.message;
    const payload = JSON.parse(message.data.sendbird);
    const localNotification = new firebase.notifications.Notification({
      show_in_foreground: true
    }).android
      .setChannelId('com.reactnativewithsendbird.default_channel_id')
      .android.setSmallIcon('sendbird_ic_notification')
      .android.setPriority(firebase.notifications.Android.Priority.High)
      .setNotificationId(message.messageId)
      .setTitle('New message')
      .setSubtitle(`Unread message: ${payload.unread_message_count}`)
      .setBody(text)
      .setData(payload);
    return firebase.notifications().displayNotification(localNotification);
  } catch (e) {
    return Promise.resolve();
  }
};
