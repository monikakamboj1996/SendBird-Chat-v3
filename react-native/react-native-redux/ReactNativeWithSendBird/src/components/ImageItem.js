import React, { Component } from 'react';
import { CachedImage } from 'react-native-cached-image';
import { Image, View } from 'react-native';

class ImageItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      width: 100,
      height: 100,
      isMounted: false
    };
  }

  componentDidMount() {
    this.setState(
      {
        isMounted: true
      },
      () => {
        Image.getSize(this.props.message, (width, height) => {
          const { resizeWidth, resizeHeight } = this._imageResize(width, height);
          if (this.state.isMounted) this.setState({ width: resizeWidth, height: resizeHeight });
        });
      }
    );
  }
  componentWillUnmount() {
    this.setState({
      isMounted: false
    });
  }

  _imageResize = (width, height) => {
    const IMAGE_MAX_SIZE = 160;
    const scaleWidth = IMAGE_MAX_SIZE / width;
    const scaleHeight = IMAGE_MAX_SIZE / height;

    let scale = scaleWidth <= scaleHeight ? scaleWidth : scaleHeight;
    if (scale > 1) {
      scale = 1;
    }

    const resizeWidth = width * scale;
    const resizeHeight = height * scale;
    return { resizeWidth: resizeWidth, resizeHeight: resizeHeight };
  };

  render() {
    return (
      <View style={{}}>
        <CachedImage
          style={{ width: this.state.width, height: this.state.height }}
          source={{ uri: this.props.message }}
          resizeMode="contain"
        />
      </View>
    );
  }
}

export { ImageItem };
