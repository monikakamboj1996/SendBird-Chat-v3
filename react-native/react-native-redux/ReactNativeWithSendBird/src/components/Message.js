import React, { Component } from 'react';
import { View, Text, TouchableWithoutFeedback } from 'react-native';
import { MessageAvatar } from './MessageAvatar';
import { MessageContainer } from './MessageContainer';
import { connect } from 'react-redux';
import { onUserMessagePress } from '../actions';

class Message extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bgColor: ''
    };
  }

  _onMessagePressed = () => {
    this.props.onUserMessagePress(this.props.message);
  };

  _containsMessageWithId = (messages, messageId) => {
    return messages.find(message => message.messageId === messageId);
  };

  componentDidUpdate(prevProps) {
    if (!prevProps.selectedMessages && !this.props.selectedMessages) {
      return;
    }
    const messageId = prevProps.message.messageId;
    if (
      this.props.selectedMessages &&
      this._containsMessageWithId(this.props.selectedMessages, messageId) &&
      (!prevProps.selectedMessages || !this._containsMessageWithId(prevProps.selectedMessages, messageId))
    ) {
      this.setState({
        bgColor: '#afceff'
      });
      return;
    }
    if (
      prevProps.selectedMessages &&
      this._containsMessageWithId(prevProps.selectedMessages, messageId) &&
      (!prevProps.selectedMessages || !this._containsMessageWithId(this.props.selectedMessages, messageId))
    ) {
      this.setState({
        bgColor: '#f1f2f6'
      });
      return;
    }
  }

  _renderMessageAvatar = () => {
    return this.props.isUser ? null : (
      <MessageAvatar isShow={this.props.isShow} uri={this.props.profileUrl} onAvatarPress={this.props.onAvatarPress} />
    );
  };

  render() {
    return (
      <TouchableWithoutFeedback onLongPress={this._onMessagePressed} delayLongPress={1500}>
        <View style={[styles.messageViewStyle, { backgroundColor: this.state.bgColor }]}>
          <View
            style={{
              flexDirection: this.props.isUser ? 'row-reverse' : 'row',
              paddingLeft: 14,
              paddingRight: 14,
              paddingTop: 4,
              paddingBottom: 4
            }}
          >
            {this._renderMessageAvatar()}
            <MessageContainer
              isShow={this.props.isShow}
              isUser={this.props.isUser}
              nickname={this.props.nickname}
              message={this.props.message}
              time={this.props.time}
              isEdited={this.props.isEdited}
              readCount={this.props.readCount}
            />
          </View>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const AdminMessage = props => {
  return (
    <TouchableWithoutFeedback onLongPress={this._onMessagePressed}>
      <View
        style={[
          styles.messageViewStyle,
          {
            padding: 8,
            marginTop: 8,
            marginBottom: 8,
            marginLeft: 14,
            marginRight: 14,
            backgroundColor: '#e6e9f0'
          }
        ]}
      >
        <Text>{props.message}</Text>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = {
  messageViewStyle: {
    transform: [{ scaleY: -1 }]
  }
};

const mapStateToProps = ({ chat }) => {
  const { selectedMessages } = chat;
  return { selectedMessages };
};

const connectedMessage = connect(
  mapStateToProps,
  {
    onUserMessagePress
  }
)(Message);

export { connectedMessage as Message, Message as TestMessage, AdminMessage };
