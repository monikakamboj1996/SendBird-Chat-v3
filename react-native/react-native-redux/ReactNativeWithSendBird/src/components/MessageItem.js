import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Icon } from 'react-native-elements';

const TextItem = props => {
  return (
    <View style={{}}>
      <Text style={{ fontSize: 12, color: props.isUser ? '#fff' : '#000' }}>{props.message}</Text>
    </View>
  );
};

class FileItem extends Component {
  constructor(props) {
    super(props);
  }

  _renderMessage() {
    if (this.props.message.length > 13) {
      return this.props.message.substring(0, 10) + '...';
    }
    return this.props.message;
  }

  render() {
    return (
      <View style={{ flexDirection: 'row' }}>
        <Icon
          containerStyle={{ marginLeft: 0 }}
          iconStyle={{ margin: 0, padding: 0 }}
          name="file"
          type="font-awesome"
          color={this.props.isUser ? '#fff' : '#000'}
          size={16}
        />
        <Text style={{ marginLeft: 8 }}>{this._renderMessage()}</Text>
      </View>
    );
  }
}

const styles = {};

export { TextItem, FileItem };
