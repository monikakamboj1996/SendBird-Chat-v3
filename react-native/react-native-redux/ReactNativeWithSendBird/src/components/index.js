export * from './Spinner';
export * from './Message';
export * from './MessageItem';
export * from './ImageItem';
export * from './MessageInput';
export * from './Hr';
export * from './MessageContainer';
export * from './MessageAvatar';
export * from './MessageBubble';
export { Button, Icon, Avatar, FormLabel, FormInput, FormValidationMessage, ListItem } from 'react-native-elements';
