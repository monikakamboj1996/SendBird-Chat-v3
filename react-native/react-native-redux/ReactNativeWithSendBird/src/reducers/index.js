import { combineReducers } from 'redux';
import login from './loginReducer';
import menu from './menuReducer';
import profile from './profileReducer';
import openChannel from './openChannelReducer';
import openChannelCreate from './openChannelCreateReducer';
import chat from './chatReducer';
import member from './memberReducer';
import blockUser from './blockUserReducer';
import groupChannel from './groupChannelReducer';
import groupChannelInvite from './groupChannelInviteReducer';

export default combineReducers({
  login,
  menu,
  profile,
  openChannel,
  openChannelCreate,
  chat,
  member,
  blockUser,
  groupChannel,
  groupChannelInvite
});

export loginReducer from './loginReducer';
export menuReducer from './menuReducer';
export profileReducer from './profileReducer';
export openChannelReducer from './openChannelReducer';
export openChannelCreateReducer from './openChannelCreateReducer';
export chatReducer from './chatReducer';
export memberReducer from './memberReducer';
export blockUserReducer from './blockUserReducer';
export groupChannelReducer from './groupChannelReducer';
export groupChannelInviteReducer from './groupChannelInviteReducer';
