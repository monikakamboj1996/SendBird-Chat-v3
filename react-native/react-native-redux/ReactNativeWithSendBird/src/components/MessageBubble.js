import React, { Component } from 'react';
import { View, Text } from 'react-native';
import Video from 'react-native-video';
import { FileItem, TextItem } from './MessageItem';
import { ImageItem } from './ImageItem';

const _isVideo = props => {
  return props.message.type.match(/^video\/.+$/);
};

const _isImage = props => {
  return props.message.type.match(/^image\/.+$/);
};

export class MessageBubble extends Component {
  constructor(props) {
    super(props);
  }

  _renderNickname = nickname => {
    return nickname ? <Text style={{ fontSize: 9, color: '#7048e8', paddingBottom: 4 }}>{nickname}</Text> : null;
  };

  _renderMessageItem = message => {
    if (message.isUserMessage()) {
      return <TextItem isUser={this.props.isUser} message={message.message} />;
    } else if (_isImage(this.props)) {
      return <ImageItem isUser={this.props.isUser} message={message.url.replace('http://', 'https://')} />;
    } else if (_isVideo(this.props)) {
      return (
        <View style={styles.videoContainer}>
          <Video source={{ uri: message.url }} style={styles.video} paused={true} controls={true} />
        </View>
      );
    } else {
      return <FileItem isUser={this.props.isUser} message={message.name} />;
    }
  };

  render() {
    const message = this.props.message;
    if (!message) {
      return null;
    }
    return (
      <View
        style={{
          maxWidth: 250,
          padding: 8,
          borderRadius: 8,
          backgroundColor: this.props.isUser ? '#5F3DC4' : '#e6e6e6'
        }}
      >
        {this.props.isUser || !this.props.isShow ? null : this._renderNickname(this.props.nickname)}
        <View style={{}}>{this._renderMessageItem(this.props.message)}</View>
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'flex-end',
            paddingLeft: 8
          }}
        >
          <Text
            style={{
              fontSize: 8,
              color: this.props.isUser ? '#E9EBEF' : '#878d99'
            }}
          >
            {this.props.time}
          </Text>
        </View>
        {this.props.isEdited && (
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'flex-end',
              paddingLeft: 8
            }}
          >
            <Text
              style={{
                fontSize: 8,
                color: this.props.isUser ? '#E9EBEF' : '#878d99'
              }}
            >
              edited
            </Text>
          </View>
        )}
      </View>
    );
  }
}

const styles = {
  videoContainer: {
    minWidth: 240,
    minHeight: 180,
    marginBottom: 4,
    borderRadius: 8
  },
  video: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    borderRadius: 8
  }
};
