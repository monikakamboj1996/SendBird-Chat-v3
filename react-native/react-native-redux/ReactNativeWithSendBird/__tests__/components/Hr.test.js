import 'react-native';
import React from 'react';
import { HR } from '../../src/components';
import renderer from 'react-test-renderer';

describe('component/Hr', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HR />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
