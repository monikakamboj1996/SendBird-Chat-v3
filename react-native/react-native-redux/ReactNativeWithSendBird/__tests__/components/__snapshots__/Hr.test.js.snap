// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Hr renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#e6e6e6",
        "flexDirection": "row",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;
